arch/risc-v/src/cmake/Toolchain.cmake

```diff
@@ -370,3 +369,13 @@ endif()
 if(CONFIG_MM_UBSAN_TRAP_ON_ERROR)
   add_compile_options(-fsanitize-undefined-trap-on-error)
 endif()
+
+execute_process(
+  COMMAND ${CMAKE_CXX_COMPILER} -Wl,--print-memory-usage 2>&1
```

Comment:
`--print-memory-usage ` is binutils feature depends on GCC version, why not check GCC version before add this option?

Reply:
If we directly use the compiler to try this compilation option, and then determine whether it is supported, avoiding the need to judge each compiler version, would this be a once-and-for-all solution?

Reply:
But it is not as fast as checking GCCVER directly, current implement need to exec a command line program and capture its output.

Reply:
OK, that looks better. Let me modify this patch.